pkg/controller/core/clusterqueue_controller.go

```diff
@@ -177,6 +177,13 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	log := ctrl.LoggerFrom(ctx)
 	log.V(2).Info("Reconcile ClusterQueue")
 
+	if features.Enabled(features.TopologyAwareScheduling) && !r.cache.TASCache().SyncedFlavors() {
+		log.V(2).Info("Waiting for TAS cache to be synced")
+		return ctrl.Result{
```

Comment:
How long in takes in practice? I think it is fair to just return error to avoid hard-coding such delays.